nemo_text_processing/text_normalization/hi/taggers/cardinal.py

```diff
@@ -0,0 +1,324 @@
+
+        # Transducer for eleven hundred -> 1100 or twenty one hundred eleven -> 2111
+        graph_hundreds_as_thousand = create_graph_suffix(teens_and_ties, suffix_hundreds, 2)
+        graph_hundreds_as_thousand |= create_larger_number_graph(teens_and_ties, suffix_hundreds, 1, digit)
+        graph_hundreds_as_thousand |= create_larger_number_graph(teens_and_ties, suffix_hundreds, 0, teens_ties)
+        self.graph_hundreds_as_thousand = graph_hundreds_as_thousand
+
+        # Thousands and Ten thousands graph
+        suffix_thousands = pynutil.insert(" हज़ार")
+        graph_thousands = create_graph_suffix(digit, suffix_thousands, 3)
+        graph_thousands |= create_larger_number_graph(digit, suffix_thousands, 2, digit)
+        graph_thousands |= create_larger_number_graph(digit, suffix_thousands, 1, teens_ties)
+        graph_thousands |= create_larger_number_graph(digit, suffix_thousands, 0, graph_hundreds)
+        graph_thousands.optimize()
+        self.graph_thousands = graph_thousands
+
+        graph_ten_thousands = create_graph_suffix(teens_and_ties, suffix_thousands, 3)
+        graph_ten_thousands |= create_larger_number_graph(teens_and_ties, suffix_thousands, 2, digit)
+        graph_ten_thousands |= create_larger_number_graph(teens_and_ties, suffix_thousands, 1, teens_ties)
+        graph_ten_thousands |= create_larger_number_graph(teens_and_ties, suffix_thousands, 0, graph_hundreds)
+        graph_ten_thousands.optimize()
+
+        # Lakhs graph and ten lakhs graph
+        suffix_lakhs = pynutil.insert(" लाख")
+        graph_lakhs = create_graph_suffix(digit, suffix_lakhs, 5)
+        graph_lakhs |= create_larger_number_graph(digit, suffix_lakhs, 4, digit)
+        graph_lakhs |= create_larger_number_graph(digit, suffix_lakhs, 3, teens_ties)
+        graph_lakhs |= create_larger_number_graph(digit, suffix_lakhs, 2, graph_hundreds)
+        graph_lakhs |= create_larger_number_graph(digit, suffix_lakhs, 1, graph_thousands)
+        graph_lakhs |= create_larger_number_graph(digit, suffix_lakhs, 0, graph_ten_thousands)
+        graph_lakhs.optimize()
+
+        graph_ten_lakhs = create_graph_suffix(teens_and_ties, suffix_lakhs, 5)
+        graph_ten_lakhs |= create_larger_number_graph(teens_and_ties, suffix_lakhs, 4, digit)
+        graph_ten_lakhs |= create_larger_number_graph(teens_and_ties, suffix_lakhs, 3, teens_ties)
+        graph_ten_lakhs |= create_larger_number_graph(teens_and_ties, suffix_lakhs, 2, graph_hundreds)
+        graph_ten_lakhs |= create_larger_number_graph(teens_and_ties, suffix_lakhs, 1, graph_thousands)
+        graph_ten_lakhs |= create_larger_number_graph(teens_and_ties, suffix_lakhs, 0, graph_ten_thousands)
+        graph_ten_lakhs.optimize()
+
+        # Crores graph ten crores graph
+        suffix_crores = pynutil.insert(" करोड़")
+        graph_crores = create_graph_suffix(digit, suffix_crores, 7)
+        graph_crores |= create_larger_number_graph(digit, suffix_crores, 6, digit)
```

Comment:
since all function calls seem to be from X to 0, would it be possible to modify the function to call just once and have it internally do the iteration?